src/apple/system.rs

```diff
@@ -450,9 +450,25 @@ impl SystemExt for System {
         get_system_info(libc::KERN_HOSTNAME, None)
     }
 
-    fn get_version(&self) -> Option<String> {
+    fn get_kernel_version(&self) -> Option<String> {
         get_system_info(libc::KERN_OSRELEASE, None)
     }
+
+    fn get_os_version(&self) -> Option<String> {
+        unsafe {
+            // get the size for the buffer first
+            let mut size = 0;
+            get_sys_value_by_name(b"kern.osproductversion\0", &mut size, std::ptr::null_mut());
+            // now create a buffer with the size and get the real value
+            let mut buf = vec![0_u8; size as usize];
+            get_sys_value_by_name(
+                b"kern.osproductversion\0",
+                &mut size,
+                buf.as_mut_ptr() as *mut c_void,
```

Comment:
```suggestion
                buf.as_mut_ptr() as *mut _,
```

Reply:
I'm not a big fan of the anonymous type conversion like this, is there a reason you want this change?